modules/library/main/src/test/java/org/geotools/filter/function/FilterFunction_arrayAnyMatchTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2019, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.function;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.geotools.filter.FilterFactoryImpl;
+import org.junit.Test;
+import org.opengis.filter.expression.Function;
+
+public class FilterFunction_arrayAnyMatchTest {
+    @Test
+    public void testMatchIntegers() throws Exception {
+        FilterFactoryImpl ff = new FilterFactoryImpl();
+        Function func =
+                ff.function(
+                        "arrayAnyMatch",
+                        ff.literal(new Integer[] {1, 2, 3, 4, 5, 6, 7, 8, 9}),
+                        ff.literal(9));
+        assertTrue((Boolean) func.evaluate(new Object()));
+    }
+
+    @Test
+    public void testMatchStrings() throws Exception {
+        FilterFactoryImpl ff = new FilterFactoryImpl();
+        Function func =
+                ff.function(
+                        "arrayAnyMatch",
+                        ff.literal(new String[] {"a", "b", "c", "d"}),
+                        ff.literal("b"));
+        assertTrue((Boolean) func.evaluate(new Object()));
+    }
+
+    @Test
+    public void testDoesNotMatchIntegers() throws Exception {
+        FilterFactoryImpl ff = new FilterFactoryImpl();
+        Function func =
+                ff.function(
+                        "arrayAnyMatch",
+                        ff.literal(new Integer[] {1, 2, 3, 4, 5, 6, 7, 8, 9}),
+                        ff.literal(10));
+        assertFalse((Boolean) func.evaluate(new Object()));
+    }
+
+    @Test
+    public void testDoesNotMatchStrings() throws Exception {
+        FilterFactoryImpl ff = new FilterFactoryImpl();
+        Function func =
+                ff.function(
+                        "arrayAnyMatch",
+                        ff.literal(new String[] {"a", "b", "c", "d"}),
+                        ff.literal("g"));
+        assertFalse((Boolean) func.evaluate(new Object()));
+    }
```

Comment:
Test the edge cases to check if type conversion is working:
```
    @Test
     public void testDoesNotMatchStrings() throws Exception {
         FilterFactoryImpl ff = new FilterFactoryImpl();
         Function func =
                 ff.function(
                         "arrayAnyMatch",
                         ff.literal("a"),
                         ff.literal("g"));
         assertFalse((Boolean) func.evaluate(new Object()));
     }
```

Can I ask how this is represented in `ECQL`?

Reply:
From my knowledge you cannot express array literals in ECQL, so this is only going to be used on datasource that have array type properties (we use it that way in the postgis driver)